cirq/google/gate_sets.py

```diff
@@ -12,116 +12,167 @@
-                op_deserializer.DeserializingArg(
-                    serialized_name='half_turns',
-                    constructor_arg_name='exponent')
-            ]),
-        op_deserializer.GateOpDeserializer(
-            serialized_gate_id='exp_z',
-            gate_constructor=ops.ZPowGate,
-            args=[
-                op_deserializer.DeserializingArg(
-                    serialized_name='half_turns',
-                    constructor_arg_name='exponent')
-            ]),
-        op_deserializer.GateOpDeserializer(
-            serialized_gate_id='exp_11',
-            gate_constructor=ops.CZPowGate,
-            args=[
-                op_deserializer.DeserializingArg(
-                    serialized_name='half_turns',
-                    constructor_arg_name='exponent')
-            ]),
-        op_deserializer.GateOpDeserializer(
-            serialized_gate_id='meas',
-            gate_constructor=ops.MeasurementGate,
-            args=[
-                op_deserializer.DeserializingArg(serialized_name='key',
-                                                 constructor_arg_name='key'),
-                op_deserializer.DeserializingArg(
-                    serialized_name='invert_mask',
-                    constructor_arg_name='invert_mask',
-                    value_func=lambda x: tuple(cast(list, x)))
-            ],
-            num_qubits_param='num_qubits'),
-    ])
+        _SYC_DESERIALIZER,
+        *SINGLE_QUBIT_DESERIALIZERS,
+        *SINGLE_QUBIT_HALF_PI_DESERIALIZERS,
+        MEASUREMENT_DESERIALIZER,
+    ],
+)
+
+# The xmon gate set.
+XMON: serializable_gate_set.SerializableGateSet = (
+    serializable_gate_set.SerializableGateSet(
+        gate_set_name='xmon',
+        serializers=[
+            op_serializer.GateOpSerializer(
+                gate_type=ops.PhasedXPowGate,
+                serialized_gate_id='exp_w',
+                args=[
+                    op_serializer.SerializingArg(
+                        serialized_name='axis_half_turns',
+                        serialized_type=float,
+                        gate_getter='phase_exponent'),
+                    op_serializer.SerializingArg(serialized_name='half_turns',
+                                                 serialized_type=float,
+                                                 gate_getter='exponent')
+                ]),
+            op_serializer.GateOpSerializer(gate_type=ops.ZPowGate,
+                                           serialized_gate_id='exp_z',
+                                           args=[
+                                               op_serializer.SerializingArg(
+                                                   serialized_name='half_turns',
+                                                   serialized_type=float,
+                                                   gate_getter='exponent')
+                                           ]),
+            op_serializer.GateOpSerializer(
```

Comment:
why doesn't this use the `common_serializers`. Otherwise they don't seem so common

Reply:
Sigh.  XMON uses a different naming convention than those serializers.  "xy" vs "exp_w", etc, and those strings are used all over the place in tests.

How about this?  I will change it over in a follow-up PR, but I think doing so in this PR would conflate the change, since XMON hasn't really changed.

Reply:
for sure

Reply:
common_gates.py:

```python
def get_single_qubit_gates(what_do_you_call_a_phasedxpow: str):
  if what_do_you_call_a_phasedxpow not in ['xy', 'exp_w']:
    raise ValueError("https://xkcd.com/927/")
  ...
```